app/src/main/java/com/ataulm/artcollector/Navigation.kt

```diff
@@ -0,0 +1,26 @@
+package com.ataulm.artcollector
+
+import android.content.Intent
+import android.net.Uri
+
+enum class Navigation(private val semiQualifiedActivityName: String) {
+
+    PAINTING("painting.ui.PaintingActivity"),
+    PAINTINGS("paintings.ui.PaintingsActivity");
+
+    fun viewIntent(artistId: String, paintingId: String): Intent {
+        val uri = Uri.Builder()
+                .scheme("https://")
+                .authority("art-collector.ataulm.com")
+                .path("$artistId/$paintingId")
+                .build()
+
+        return viewIntent().setData(uri)
+    }
+
+    fun viewIntent(): Intent {
+        val componentName = "${BuildConfig.APPLICATION_ID}.$semiQualifiedActivityName"
+        return Intent(Intent.ACTION_VIEW)
+                .setClassName(BuildConfig.APPLICATION_ID, componentName)
```

Comment:
this shouldn't be necessary with App Links if I understand correctly but we'll see. This is to avoid having the base module dependent on the feature module. ie. this module doesn't really know about the new feature that we just added.

it'd just be opening the URL `https://art-collector.ataulm.com/artistId/paintingId` and this navigator should not really care who handles that URL: browser/another app/another activity in this app etc.